### PR TITLE
ogg: Add option to disable testing

### PR DESCRIPTION
**Pull request by gogoprog**
Specify library name and version:  **ogg**

Sometimes you don't want to run the BUILD_TESTING steps from the CMakeLists.
This adds an option for it

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ogg/all/conanfile.py [hunk 1]
```diff
@@ -16,10 +16,12 @@ class OggConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "testing": [True, False],
```
Comment (uilianries) [inline recipes/ogg/all/conanfile.py:19]:
```suggestion
        "build_testing": [True, False],
```

File: recipes/ogg/all/conanfile.py [hunk 2]
```diff
@@ -16,10 +16,12 @@ class OggConan(ConanFile):
...
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "testing": True,
```
Comment (uilianries) [inline recipes/ogg/all/conanfile.py:24]:
```suggestion
        "build_testing": False,
```

File: recipes/ogg/all/conanfile.py [hunk 3]
```diff
@@ -58,6 +60,8 @@ def _configure_cmake(self):
         self._cmake = CMake(self)
         # Generate a relocatable shared lib on Macos
         self._cmake.definitions["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        if not self.options.testing:
+            self._cmake.definitions["BUILD_TESTING"] = False
```
Comment (uilianries) [inline recipes/ogg/all/conanfile.py:64]:
```suggestion
        self._cmake.definitions["BUILD_TESTING"] = self.options.build_testing
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution!
- Conan Center does not distribute testing neither example by default
- Try to use the same option name, so users don't need to think

**Comment by gogoprog** [comment]
Hello,
I agree with the suggestions and committed them.

Regarding the build_testing default to false, I just wanted to point out that the previous behavior was always building the test.
Again I also agree it should not build by default.

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
Why this option? We should disable tests unconditionally like all other CCI recipes. Moreover this option doesn't change binaries but it contributes to package id.

**Comment by gogoprog** [comment]
@SpaceIm I did not want to change the existing behavior which was building the test. I am not against removing the option and never build the test

**Comment by jgsogo** [comment]
We have finally documented the way to handle tests: https://github.com/conan-io/conan-center-index/pull/11588

**Pull request merged**